### PR TITLE
Test against develop

### LINKED ISSUE #91: Test against develop
We are testing against custom branch now, we should switch back to testing against develop as most of the fixes are in the develop now.

### PR DESCRIPTION
**Pull request by kamils-iRonin**
closes #91

### CODE REVIEW

File: lib/foreman_puppet.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module ForemanPuppet
-  FOREMAN_EXTRACTION_VERSION = '2.6'.freeze
+  FOREMAN_EXTRACTION_VERSION = '2.7'.freeze
```
Comment (ezr-ondrej) [inline lib/foreman_puppet.rb:2]: Why? This will be correct as once we extract puppet 2.6 => 3.0 and 3.0 >= 2.7, but it just looks weird.
We can go with 3.0 directly :)

File: app/models/concerns/foreman_puppet/extensions/host.rb [hunk 2]
```diff
@@ -9,6 +9,9 @@ module Host
         env_assoc = reflect_on_association(:environment)
         env_assoc&.instance_variable_set(:@class_name, 'ForemanPuppet::Environment')
 
+        host_classes_assoc = reflect_on_association(:host_classes)
+        host_classes_assoc&.instance_variable_set(:@class_name, 'ForemanPuppet::HostClass')
```
Comment (ezr-ondrej) [inline app/models/concerns/foreman_puppet/extensions/host.rb:13]: both should be `unless extracted`

File: app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb [hunk 3]
```diff
@@ -28,6 +28,10 @@ def destroy
           process_response @hostgroup_class.destroy_all
         end
 
+        def resource_class
+          ForemanPuppet::HostgroupClass
```
Comment (ezr-ondrej) [inline app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb:32]: Why are these needed? :thinking:

**Review by ezr-ondrej — COMMENTED** [review]
Few nits, but apart of that great job!
Do we want to merge before release for 2.5?

File: app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb (hunk 3 above)
Comment (kamils-iRonin) [reply app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb:32]: to return `ForemanPuppet::HostgroupClass` instead of `::HostgroupClass` from core

```ruby
From: /Users/kamilszubrycht/Documents/Development/foreman_puppet/app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb:33 ForemanPuppet::Api::V2::HostgroupClassesController#resource_class:

    31: def resource_class
    32:   binding.pry
 => 33:   ForemanPuppet::HostgroupClass
    34: end

[1] pry(#<ForemanPuppet::Api::V2::HostgroupClassesController>)> super
=> HostgroupClass(id: integer, hostgroup_id: integer, puppetclass_id: integer, hostgroup_puppet_facet_id: integer)
[2] pry(#<ForemanPuppet::Api::V2::HostgroupClassesController>)> ForemanPuppet::HostgroupClass
=> ForemanPuppet::HostgroupClass(id: integer, hostgroup_id: integer, puppetclass_id: integer, hostgroup_puppet_facet_id: integer)
```

File: lib/foreman_puppet.rb (hunk 1 above)
Comment (kamils-iRonin) [reply lib/foreman_puppet.rb]: bumped into `3.0`, on current develop `SETTINGS[:version].notag` returns `2.6.0`

**Comment by kamils-iRonin** [comment]
> Do we want to merge before release for 2.5?

I think it makes sense to merge it before release for 2.5

File: .github/workflows/ruby_tests.yml [hunk 4]
```diff
@@ -66,7 +66,7 @@ jobs:
         run: bundle exec rake spec
   test_ruby:
     runs-on: ubuntu-latest
-    needs: rubocop
+    # needs: rubocop
```
Comment (ezr-ondrej) [inline .github/workflows/ruby_tests.yml:69]: Yeah, I've got no idea what's wrong with rubocop :(

File: app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb (hunk 3 above)
Comment (ezr-ondrej) [reply app/controllers/foreman_puppet/api/v2/hostgroup_classes_controller.rb:32]: oh right, in API we have wrong namespace, so it doesn't get guessed correctly.

File: .github/workflows/ruby_tests.yml (hunk 4 above)
Comment (kamils-iRonin) [reply .github/workflows/ruby_tests.yml:69]: I have no idea either, I'm investigating

Comment (ezr-ondrej) [reply .github/workflows/ruby_tests.yml:69]: I'll take a look into that in follow-up

**Review by ezr-ondrej — APPROVED** [review]
Thanks @kamils-iRonin ! :+1:
